module/VuFind/src/VuFind/Controller/AbstractBase.php

```diff
@@ -736,16 +735,44 @@ protected function normalizeUrlForComparison($url)
         return trim(end($parts), '/');
     }
 
+    /**
+     * Checks if a followup url is set
+     *
+     * @return bool
+     */
+    protected function hasFollowupUrl()
+    {
+        return null !== $this->followup()->retrieve('url');
+    }
+
     /**
      * Retrieve a referer to keep post-login redirect pointing
      * to an appropriate location.
      * Unset the followup before returning.
      *
+     * @param bool $checkRedirect Whether the query should be checked for param 'redirect'
+     *
      * @return string
      */
-    protected function getFollowupUrl()
-    {
-        return $this->followup()->retrieve('url', '');
+    protected function getAndClearFollowupUrl($checkRedirect = false)
+    {
+        if ($url = $this->followup()->retrieveAndClear('url', '')) {
+            $lightboxParent = $this->followup()->retrieveAndClear('lightboxParent');
```

Comment:
Move this further down inside the `if (!$checkRedirect...)` block since it's not needed outside of it.

Reply:
This command also clears the lightboxParent. If it is moved inside of the if block it might cause problems later on.